release/src/release-notes.ts

```diff
@@ -118,97 +120,97 @@ enum ProductCategory {
-  if (hasCategory(labels, "Querying")) return ProductCategory.querying;
-  if (hasCategory(labels, "Reporting")) return ProductCategory.reporting;
-  if (hasCategory(labels, "Visualization"))
-    return ProductCategory.visualization;
-
-  return ProductCategory.other;
+  return category ?? ProductCategory.other;
 };
 
 // Format issues for a single product category
-const formatCategoryIssues = (category: string, issues: Issue[]): string => {
-  return `**${category}**\n\n${issues.map(formatIssue).join("\n")}`;
+const formatIssueCategory = (categoryName: ProductCategory, issues: Issue[]): string => {
+  return `**${categoryName}**\n\n${issues.map(formatIssue).join("\n")}`;
 };
 
 // We want to alphabetize the issues by product category, with "Other" (uncategorized) issues as the caboose
-const sortCategories = (categories: ProductCategory[]): ProductCategory[] => {
+const sortCategories = (categories: ProductCategory[]) => {
   const uncategorizedIssues = categories.filter(
     category => category === ProductCategory.other,
   );
-  return categories
+  const sortedCategories = categories
     .filter(cat => cat !== ProductCategory.other)
-    .sort((a, b) => a.localeCompare(b))
-    .concat(uncategorizedIssues);
+    .sort((a, b) => a.localeCompare(b));
+
+  return [
+    ...sortedCategories,
+    ...uncategorizedIssues,
+  ];
 };
 
 // For each issue category ("Enhancements", "Bug Fixes", etc.), we want to group issues by product category
-const groupIssuesByProductCategory = (issues: Record<string, Issue[]>) => {
+const formatIssues = (issueMap: CategoryIssueMap): string => {
   const categories = sortCategories(
-    Object.keys(issues).map(key => key as ProductCategory),
+    Object.keys(issueMap) as ProductCategory[],
   );
 
   return categories
-    .map(category => formatCategoryIssues(category, issues[category]))
+    .map(categoryName => formatIssueCategory(categoryName, issueMap[categoryName]))
     .join("\n\n");
 };
 
 export const categorizeIssues = (issues: Issue[]) => {
-  const issueMap = {
-    [IssueType.bugFixes]: {},
-    [IssueType.enhancements]: {},
-    [IssueType.alreadyFixedIssues]: {},
-    [IssueType.underTheHoodIssues]: {},
-  };
-
-  issues
+  return issues
     .filter(issue => !isHiddenIssue(issue))
-    .forEach(issue => {
+    .reduce((issueMap: IssueMap, issue: Issue) => {
       const issueType = getIssueType(issue);
       const productCategory = getProductCategory(issue);
-      addIssueToCategory(issueMap, issue, issueType, productCategory);
-    });
 
-  return issueMap;
+      return {
+        ...issueMap,
+        [issueType]: {
+          ...issueMap[issueType],
+          [productCategory]: [
+            ...issueMap[issueType][productCategory] ?? [],
+            issue,
+          ],
+        },
+      };
+    }, {
+      [IssueType.bugFixes]: {},
+      [IssueType.enhancements]: {},
+      [IssueType.alreadyFixedIssues]: {},
+      [IssueType.underTheHoodIssues]: {},
+    } as IssueMap);
 };
```

Comment:
we prefer to use `.reduce()` over `.forEach` because we can avoid mutating an object, which often causes difficult-to-debug side-effects.